### PR TITLE
[iOS] Fix BackgroundColor property issue on SearchBar

### LINKED ISSUE #6555: SearchBar background not changing on iOS
### Description

SearchBar has a black background kind of frame and setting BackgroundColor doesn't change it:

How it should work on Android:

image

How it doesn't work on IOS:

image

### Steps to Reproduce

1. Create a new MAUI app
2. Add a SearchBar
3. Set the background color.
4. See it not work.

### Version with bug

Release Candidate 2 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

ios 15

### Did you find any workaround?

None, I tried using the handler:
```csharp
Microsoft.Maui.Handlers.SearchBarHandler.Mapper.AppendToMapping("NoBackground", (h, v) =>
        {
            h.PlatformView.BackgroundColor = Colors.Transparent.ToPlatform();
        });
```
But it didn't work.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix BackgroundColor property issue on iOS SearchBar.

### Issues Fixed

Fixes #6555

### CODE REVIEW

**Comment by rmarinho** [comment]
You need to update the Unshipped api, you can use the IDE to do that

**Comment by jsuarezruiz** [comment]
> You need to update the Unshipped api, you can use the IDE to do that

Updated, waiting the Build.

**Comment by hartez** [comment]
@jsuarezruiz Needs to be targeted to net6.0 branch.

File: src/Core/src/Handlers/SearchBar/SearchBarHandler.iOS.cs [hunk 1]
```diff
@@ -20,6 +21,9 @@ protected override MauiSearchBar CreatePlatformView()
 			else
 				_editor = searchBar.FindDescendantView<UITextField>();
 
+
+			_defaultTintColor = searchBar.TintColor;
```
Comment (mattleibow) [inline src/Core/src/Handlers/SearchBar/SearchBarHandler.iOS.cs:25]: Yay! Boo! Defaults!

**Review by mattleibow — APPROVED** [review]

File: src/Core/src/Platform/iOS/SearchBarExtensions.cs [hunk 2]
```diff
@@ -1,10 +1,29 @@
...
+		// TODO: NET7 maybe make this public?
+		internal static void UpdateBackground(this UISearchBar uiSearchBar, ISearchBar searchBar, UIColor? defaultTintColor = null)
+		{
+			ViewExtensions.UpdateBackground(uiSearchBar, searchBar);
+
+			var background = searchBar.Background;
+
+			if (background is SolidPaint solidPaint)
+				uiSearchBar.BarTintColor = solidPaint.Color.ToPlatform(defaultTintColor ?? ColorExtensions.BackgroundColor);
+		}
```
Comment (mattleibow) [inline src/Core/src/Platform/iOS/SearchBarExtensions.cs:25]: Is this the correct way to change colors? How would one go about changing the text box background color? Or are they all related. What if the user wants to have the "background" a color or transparent, and the search box a different color? Should we be doing a minimal style?

Comment (rmarinho) [reply src/Core/src/Platform/iOS/SearchBarExtensions.cs:25]: What do we do if the color is null?

File: src/Core/src/Platform/iOS/SearchBarExtensions.cs [hunk 3]
```diff
@@ -1,10 +1,29 @@
...
+		// TODO: NET7 maybe make this public?
+		internal static void UpdateBackground(this UISearchBar uiSearchBar, ISearchBar searchBar)
+		{
+			uiSearchBar.UpdateBackground(searchBar, null);
+		}
+
+		// TODO: NET7 maybe make this public?
+		internal static void UpdateBackground(this UISearchBar uiSearchBar, ISearchBar searchBar, UIColor? defaultTintColor = null)
+		{
+			ViewExtensions.UpdateBackground(uiSearchBar, searchBar);
```
Comment (rmarinho) [inline src/Core/src/Platform/iOS/SearchBarExtensions.cs:19]: i don't think this is needed

**Review by rmarinho — APPROVED** [review]

**Pull request merged**